### PR TITLE
Add docs and links to appropriate papers

### PR DESCRIPTION
**Pull request by adelbertc**

### CODE REVIEW

**Comment by fthomas** [comment]
Nice!

**Comment by julien-truffaut** [comment]
good idea

File: core/src/main/scala/cats/Applicative.scala [hunk 1]
```diff
@@ -7,8 +7,8 @@ import simulacrum._
  *
  * Allows application of a function in an Applicative context to a value in an Applicative context
  *
- * See: http://www.cs.ox.ac.uk/jeremy.gibbons/publications/iterator.pdf
- * Also: http://staff.city.ac.uk/~ross/papers/Applicative.pdf
+ * See: [[http://www.cs.ox.ac.uk/jeremy.gibbons/publications/iterator.pdf The Essence of the Iterator Pattern]]
```
Comment (julien-truffaut) [inline core/src/main/scala/cats/Applicative.scala]: this link doesn't work for me but this one does: http://www.cs.ox.ac.uk/jeremy.gibbons/publications/iterator.pdf

File: core/src/main/scala/cats/Applicative.scala [hunk 2]
```diff
@@ -7,8 +7,8 @@ import simulacrum._
...
+ * Also: [[http://staff.city.ac.uk/~ross/papers/Applicative.pdf Applicative programming with effects]]
```
Comment (julien-truffaut) [inline core/src/main/scala/cats/Applicative.scala]: same for this one: http://staff.city.ac.uk/~ross/papers/Applicative.pdf

**Comment by stew** [comment]
I would wish to use https wherever that works, which seems to work for most. but that is a minor nit.

**Comment by ceedubs** [comment]
Yeah I think we might as well go for https where it works like @stew said. Once that's in, this gets a :+1: from me. (I'm not experiencing the same issues @julien-truffaut has reported.)

**Comment by adelbertc** [comment]
@julien-truffaut Could not reproduce on my end :( Also tried just copy/pasting your links and they seem to be identical with the ones I had.

**Comment by non** [comment]
:+1:

**Comment by ceedubs** [comment]
:+1:

**Pull request merged**